source/extensions/common/wasm/null/sample_plugin/plugin.cc

```diff
@@ -45,7 +46,24 @@ FilterDataStatus PluginContext::onRequestBody(size_t body_buffer_length, bool /*
 
 void PluginContext::onLog() {
   auto path = getRequestHeader(":path");
-  logWarn("onLog " + std::to_string(id()) + " " + std::string(path->view()));
+  if (path->view() == "/test_context") {
```

Comment:
I'm not a fan of mixing fake behavior required by tests in production code. Is there any reason why this cannot be created/configured/mocked in tests? I know this is a sample plugin, but people tend to copy&paste those.

At the very least, please make the path a bit more unique (`/test_<random sha>` would be acceptable).

Reply:
I don't like this, too, but that's how the other tests do it. We need more mocks I think.